x-pack/plugins/observability_solution/apm/server/lib/apm_telemetry/collect_data_telemetry/tasks.ts

```diff
@@ -898,193 +969,353 @@ export const tasks: TelemetryTask[] = [
-            sort: {
-              '@timestamp': 'desc',
+          },
+        },
+        [SERVICE_FRAMEWORK_VERSION]: {
+          terms: {
+            field: SERVICE_FRAMEWORK_VERSION,
+            size,
+          },
+        },
+        [SERVICE_LANGUAGE_NAME]: {
+          terms: {
+            field: SERVICE_LANGUAGE_NAME,
+            size,
+          },
+          aggs: {
+            [SERVICE_LANGUAGE_VERSION]: {
+              terms: {
+                field: SERVICE_LANGUAGE_VERSION,
+                size,
+              },
             },
-            aggs: {
-              [AGENT_ACTIVATION_METHOD]: {
-                terms: {
-                  field: AGENT_ACTIVATION_METHOD,
-                  size,
-                },
+          },
+        },
+        [SERVICE_LANGUAGE_VERSION]: {
+          terms: {
+            field: SERVICE_LANGUAGE_VERSION,
+            size,
+          },
+        },
+        [SERVICE_RUNTIME_NAME]: {
+          terms: {
+            field: SERVICE_RUNTIME_NAME,
+            size,
+          },
+          aggs: {
+            [SERVICE_RUNTIME_VERSION]: {
+              terms: {
+                field: SERVICE_RUNTIME_VERSION,
+                size,
               },
-              [AGENT_VERSION]: {
-                terms: {
-                  field: AGENT_VERSION,
-                  size,
+            },
+          },
+        },
+        [SERVICE_RUNTIME_VERSION]: {
+          terms: {
+            field: SERVICE_RUNTIME_VERSION,
+            size,
+          },
+        },
+      };
+
+      const agentDataWithoutOtel = await AGENT_NAMES_WITHOUT_OTEL.reduce(
+        async (prevJob, agentName) => {
+          const data = await prevJob;
+
+          const response = await telemetryClient.search({
+            index: [indices.error, indices.metric, indices.transaction],
+            body: {
+              track_total_hits: false,
+              size: 0,
+              timeout,
+              query: {
+                bool: {
+                  filter: [{ term: { [AGENT_NAME]: agentName } }, range1d],
                 },
               },
-              [SERVICE_FRAMEWORK_NAME]: {
-                terms: {
-                  field: SERVICE_FRAMEWORK_NAME,
-                  size,
-                },
-                aggs: {
-                  [SERVICE_FRAMEWORK_VERSION]: {
-                    terms: {
-                      field: SERVICE_FRAMEWORK_VERSION,
-                      size,
-                    },
-                  },
-                },
+              sort: {
+                '@timestamp': 'desc',
               },
-              [SERVICE_FRAMEWORK_VERSION]: {
-                terms: {
-                  field: SERVICE_FRAMEWORK_VERSION,
-                  size,
-                },
+              aggs: agentNameAggs,
+            },
+          });
+
+          const { aggregations } = response;
+
+          if (!aggregations) {
+            return data;
+          }
+
+          return {
```

Comment:
This code existed before but it could be a good opportunity to review the spread operator in this reduce.